src/vsphere_cpi/spec/integration/human_readable_names_spec.rb

```diff
@@ -0,0 +1,134 @@
+    VSphereCloud::Cloud.new(options)
+  end
+
+  let(:machine_name_cpi) do
+    options = cpi_options({'enable_human_readable_name' => false})
+    VSphereCloud::Cloud.new(options)
+  end
+
+  context 'When enable human readable names is set to false' do
+    let(:environment){
+      {'bosh' => { 'groups' => ['fake-director-name', 'fake-deployment-name', 'fake-instance-group-name'] }
+      }
+    }
+    it 'create vm with UUID based name' do
+      begin
+        test_vm_id = machine_name_cpi.create_vm(
+          'agent-007',
+          @stemcell_id,
+          vm_type,
+          network_spec,
+          [],
+          environment
+        )
+        expect(test_vm_id).to_not be_nil
+        expect(test_vm_id.size).to eq 39
+        expect(test_vm_id).to start_with('vm')
+      ensure
+        delete_vm(machine_name_cpi, test_vm_id)
+      end
+    end
+  end
+
+  context 'when enable human readable names is set to true' do
+    context 'when instance group name and deployment name are in ASCII characters' do
+      context 'when both instance group name and deployment name are set' do
+        let(:environment){
+          {'bosh' => { 'groups' => ['fake-director-name', 'fake-deployment-name', 'fake-instance-group-name'] }
+          }
+        }
+        it 'create vm with human readable name' do
+          begin
+            test_vm_id = human_readable_name_cpi.create_vm(
+                'agent-007',
+                @stemcell_id,
+                vm_type,
+                network_spec,
+                [],
+                environment
+            )
+            expect(test_vm_id).to_not be_nil
+            expect(test_vm_id.size).to eq 58
+            expect(test_vm_id).to start_with('fake-instance-group-name_fake-deployment-name_')
+          ensure
+            delete_vm(human_readable_name_cpi, test_vm_id)
+          end
+        end
+      end
+
+      context 'when bosh environment metadata is not in correct format' do
+        let(:environment){
+          {'bosh' => { 'groups' => ['fake-director-name', 'fake-deployment-name'] }
+          }
+        }
+        it 'create vm with UUID based name' do
+          begin
+            test_vm_id = human_readable_name_cpi.create_vm(
+              'agent-007',
+              @stemcell_id,
+              vm_type,
+              network_spec,
+              [],
+              environment
+            )
+            expect(test_vm_id).to_not be_nil
+            expect(test_vm_id.size).to eq 39
+            expect(test_vm_id).to start_with('vm')
+          ensure
+            delete_vm(human_readable_name_cpi, test_vm_id)
+          end
+        end
+      end
+    end
+
+    context 'when instance group name and deployment name contain non_ASCII characters' do
+      let(:environment){
+        {'bosh' => { 'groups' => ['fake-director-name', 'ÅÅÅÅ', 'αβγδεζηθικλμνξοπρςστυφχψω'] }
+        }
+      }
+      it 'create vm with human readable name' do
```

Comment:
Looks like the `it` examples in this file fall into only two categories:

1. It creates a VM with a human readable name
2. It creates a VM with a UUID based name

We probably want RSpec's shared examples in that case:

https://relishapp.com/rspec/rspec-core/docs/example-groups/shared-examples

Reply:
Done. I have implemented with shared examples.  Use structure like this:

when human readable name set to false
  normal way (no shared example used since only 1 test)
when set to true
  shared_example (create vm with human readable name)  
  shared_example (create vm with uuid based name)

  context 'when bosh environment metadata is not in correct format' do => uuid way
  context 'when inst_grp and deployment names contain non_ASCII characters'  => uuid way
  context 'when both names are set with ASCII characters only'  => human readable way